OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -155,7 +155,8 @@ public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr, World world)
 					cells.Add(t, isCloseEnough && world.IsCellBuildable(t, buildingInfo) && res.GetResource(t) == null);
 			}
 
-			var pal = wr.Palette("terrain");
+			var pbi = producer.Info.Traits.Get<PlaceBuildingInfo>();
```

Comment:
You should cache this in the OG constructor instead of querying it every render frame.